worker/sshserver/sshauth/auth.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshauth
+
+import (
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	"github.com/juju/names/v5"
+	jujussh "github.com/juju/utils/v3/ssh"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/juju/juju/state"
+)
+
+// Logger holds the methods required to log messages.
+type Logger interface {
+	Errorf(string, ...interface{})
+}
+
+// Authenticator is the struct to authorize users' ssh connection.
+type Authenticator struct {
+	statePool *state.StatePool
+	logger    Logger
+}
+
+// NewAuthenticator create an Authorizer. It takes as an argument the StatePool and a Logger.
```

Comment:
Suggestion to avoid writing what is already available in the function signature.
```suggestion
// NewAuthenticator create an Authorizer it uses the system state to search for a user's public key.
```

Reply:
this godoc goes old superfast, becuase now you need to "list" functionalities in the constructor

Reply:
That's fine, we can remove the second half and just leave it as "creates an Authorizer". The godoc for what an authorizer does should be left to the Authorizer.
But still remove the sentence about the arguments because that's evident from the signature.